starter-code/src/clue.js

```diff
@@ -1,94 +1,139 @@
+var missScarlet = new Character('Kasandra', 'Scarlet', 'red', 'She is an A-list movie star with a dark past', 31, 'https://metrouk2.files.wordpress.com/2016/07/miss-scarlett.jpg', 'Actor');
+var mrsPeacock = new Character('Eleanor', 'Peacock', 'blue', 'She is from a wealthy family and uses her status and money to earn popularity', 36, 'https://metrouk2.files.wordpress.com/2016/07/mrs-peacock.jpg' , 'Socialité');
+var mrMustard = new Character('Jack', 'Mustard', 'yellow', 'He is a former football player who tries to get by on his former glory', 62, 'https://metrouk2.files.wordpress.com/2016/07/colonel-mustard.jpg', 'Retired Football player');
+
 
 // Weapons
 
-name: rope        ---   weight: 10
-name: knife       ---   weight: 8
-name: candlestick ---   weight: 2
-name: dumbbell    ---   weight: 30
-name: poison      ---   weight: 2
-name: axe         ---   weight: 15
-name: bat         ---   weight: 13
-name: trophy      ---   weight: 25
-name: pistol      ---   weight: 20
+function Weapon (name,weight){
+  this.name = name;
+  this.weight = name;
+}
+
+var rope = new Weapon ('rope', 10);
+var knife = new Weapon ('knife', 8);
+var candlestick = new Weapon ('candelstick', 2);
+var dumbbell = new Weapon ('dumbbell', 30);
+var poison = new Weapon ('poison', 2);
+var axe = new Weapon ('axe', 15);
+var bat = new Weapon ('bat', 13);
+var trophy = new Weapon ('trophy', 25);
+var pistol = new Weapon ('pistol', 20);
+
 
 // Rooms
-name: Dinning Room
-name: Conservatory
-name: Kitchen
-name: Study
-name: Library
-name: Billiard Room
-name: Lounge
-name: Ballroom
-name: Hall
-name: Spa
-name: Living Room
-name: Observatory
-name: Theater
-name: Guest House
-name: Patio
-*/
+
+function Room (name){
+  this.name = name;
+}
+
+var dinningRoom = new Room('Dinning Room');
+var conservatory = new Room('Conservatory');
+var kitchen = new Room('Kitchen');
+var library = new Room('Library');
+var billiardRoom = new Room('Billiard');
+var lounge = new Room('Lounge');
+var ballroom = new Room('Ballroom');
+var hall = new Room('Hall');
+var spa = new Room('Spa');
+var livingRoom = new Room('Living Room');
+var observatory = new Room('Observatory');
+var theater = new Room('Theater');
+var guestHouse = new Room('Guest House');
+var patio = new Room('Patio');
 
 // Characters Collection
 var charactersArray = [];
+charactersArray.push(mrGreen,mrMustard,drOrchid, profPlum,missScarlet,mrsPeacock);
 
 // Rooms' Collection
 var roomsArray = [];
+roomsArray.push(dinningRoom,conservatory,kitchen,library,billiardRoom,lounge,ballroom,hall,spa,livingRoom,observatory,theater,guestHouse,patio);
+
 
 // Weapons Collection
 var weaponsArray = [];
+weaponsArray.push(rope,knife,candlestick,dumbbell,poison,axe,bat,trophy,pistol);
+
+//method to get a Random selector
+function randomSelector (array) {
+
+  var selector;
+  var minNum = 0;
+  var maxNum = array.length;
+  if (array.length > 0) {
+    selector = Math.floor(Math.random()*(array.length - minNum))+ minNum;
```

Comment:
selector is not declared

Reply:
```
selector = Math.floor(Math.random()*(array.length + 1));
```

Reply:
Oops! Bad correction! If you are using the number as an index it's okay :disappointed: sorry